collector/report.go

```diff
@@ -0,0 +1,331 @@
+	}
+	if stringInSlice(a.Format, supportedFormats) != true {
+		return fmt.Errorf("invalid \"format\" field, %s", a.Format)
+	}
+	return nil
+}
+
+// IncomingFilename determines the filename of an incoming report
+func (a *ActiveReport) IncomingFilename() string {
+	if stringInSlice(a.Format, supportedFormats) != true {
+		// Defensive coding
+		panic("an unexpected value for format was found. Bailing...")
+	}
+	return fmt.Sprintf("%s.%s",
+		a.ReportID,
+		a.Format)
+}
+
+// SyncFilename creates a filename in the sync directory of the active report
+func (a *ActiveReport) SyncFilename() (string, error) {
+	// Defensive coding.
+	if err := a.Validate(); err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%s-%s-%s-%s-%s-probe-0.2.0.%s",
+		a.CreationTime.Format(TimestampFormat),
+		a.TestName,
+		a.ReportID,
+		a.ProbeASN,
+		a.ProbeCC,
+		a.Format), nil
+}
+
+// BackendExtra is serverside extra metadata
+type BackendExtra struct {
+	SubmissionTime time.Time `json:"submission_time" bson:"submission_time"`
+	MeasurementID  string    `json:"measurement_id" bson:"measurement_id"`
+	ReportID       string    `json:"report_id" bson:"report_id"`
+}
+
+// MeasurementEntry is the structure of measurements submitted by an OONI Probe client
+type MeasurementEntry struct {
+	// These values are added by the pipeline
+	// BucketDate     string `json:"bucket_date"`
+	// ReportFilename string `json:"report_filename"`
+
+	ID                   string       `json:"id"`
+	ReportID             string       `json:"report_id"`
+	TestName             string       `json:"test_name"`
+	TestVersion          string       `json:"test_version"`
+	MeasurementStartTime string       `json:"measurement_start_time,omitempty"`
+	TestStartTime        string       `json:"test_start_time"` // XXX these should actually be time
+	Annotations          interface{}  `json:"annotations"`
+	BackendExtra         BackendExtra `json:"backend_extra"`
+	BackendVersion       string       `json:"backend_version"`
+	DataFormatVersion    string       `json:"data_format_version"`
+	Input                string       `json:"input"`
+	InputHashes          []string     `json:"input_hashes"`
+	Options              []string     `json:"options"`
+	ProbeASN             string       `json:"probe_asn"`
+	ProbeCC              string       `json:"probe_cc"`
+	ProbeCity            string       `json:"probe_city"`
+	ProbeIP              string       `json:"probe_ip"`
+	SoftwareName         string       `json:"software_name"`
+	SoftwareVersion      string       `json:"software_version"`
+	TestHelpers          interface{}  `json:"test_helpers"`
+	TestKeys             interface{}  `json:"test_keys"`
+	TestRuntime          float64      `json:"test_runtime"`
+}
+
+// CreateNewReport creates a new report
+func CreateNewReport(store *Storage, format string) (string, error) {
+	activeReport := NewActiveReport(format)
+	// XXX put the report metadata into an activeReports in memory map
```

Comment:
What does this `XXX` mean?

Reply:
I believe this is no longer relevant (it is added to the `activeReport` map below), but I will double check that.

Reply:
Thanks. Please remove the comment if it's outdated.